pkg/util/timeutil/time_zone_util.go

```diff
@@ -66,3 +67,35 @@ func ParseFixedOffsetTimeZone(location string) (offset int, origRepr string, suc
 	}
 	return offset, strings.TrimSuffix(strings.TrimPrefix(origRepr, "("), ")"), true
 }
+
+// timeZoneOffsetStringConversion converts a string like GMT+08:00 or -08:00 to offset seconds
+// Supported time zone strings :- UTC+08:00, GMT-12:00, +05:00, UTC+08:30
+// Unsupported time zone strings :- UTC+8:00, -8:00, -12:53, UTC+08:08
+func TimeZoneOffsetStringConversion(s string) (offset int64, ok bool) {
+	pattern := `(?mi)(GMT|UTC)[+-](((12:00)|(11:(0|3)0))|(0([0-9]):(0|3)0))\b`
```

Comment:
I think this regex is more restrictive than postgres:
* It doesn't support "simpler" formats/formats missing leading zeroes or trailing zeroes -- e.g. `SET TIME ZONE 'GMT+3'` should work.
* It only supports strict 00 and 30 minute offsets -- I think postgres allows minutely, e.g. `SET TIME ZONE 'UTC+3:15'` -- or even seconds -- `SET TIME ZONE 'UTC+3:15:00'`.

Furthermore, it'd be great if you could add some unit tests as well!

Reply:
Thanks for the review. I didn't know postgres allows minute offset.
I'll make the changes

Reply:
@otan This regex wont allow +/-12:30 as input. Should that also be allowed?
I was referring the timezone table in https://www.postgresql.org/docs/7.2/timezones.html

Reply:
I think when I was playing with the postgres shell it supported up to `+- 15:59`. Ideally we should tolerate the same (including `+- 12:30`).

Reply:
Okay. Sounds good!

Reply:
I've made the changes. Added relaxed regex that allows the required inputs and also added unit test.